extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -572,55 +574,100 @@ static Match validateNestedExpressions(TemplateAnalysis templateAnalysis, ClassI
+                            if (member != null) {
+                                membersUsed
+                                        .add(member.kind() == Kind.FIELD ? member.asField().name() : member.asMethod().name());
+                            }
+                        }
                     }
-                } else if (info.isProperty()) {
-                    member = findProperty(info.asProperty().name, match.clazz(), index);
-                    if (member != null) {
-                        membersUsed.add(member.kind() == Kind.FIELD ? member.asField().name() : member.asMethod().name());
+                    if (member == null) {
+                        // Then try to find an etension method
+                        member = findTemplateExtensionMethod(info, match.type(), templateExtensionMethods, expression,
+                                index,
+                                templateIdToPathFun, results);
                     }
-                }
-                if (member == null) {
-                    // Then try to find an etension method
-                    member = findTemplateExtensionMethod(info, match.clazz(), templateExtensionMethods, expression,
-                            index,
-                            templateIdToPathFun, results);
-                }
-
-                if (member == null) {
-                    // Test whether the validation should be skipped
-                    TypeCheck check = new TypeCheck(info.isProperty() ? info.asProperty().name : info.asVirtualMethod().name,
-                            match.clazz(),
-                            info.part.isVirtualMethod() ? info.part.asVirtualMethod().getParameters().size() : -1);
-                    if (isExcluded(check, excludes)) {
-                        LOGGER.debugf(
-                                "Expression part [%s] excluded from validation of [%s] against class [%s]",
-                                info.value,
-                                expression.toOriginalString(), match.clazz());
-                        match.clearValues();
-                        break;
+                    if (member == null) {
+                        // Test whether the validation should be skipped
+                        TypeCheck check = new TypeCheck(
+                                info.isProperty() ? info.asProperty().name : info.asVirtualMethod().name,
+                                match.clazz(),
+                                info.part.isVirtualMethod() ? info.part.asVirtualMethod().getParameters().size() : -1);
+                        if (isExcluded(check, excludes)) {
+                            LOGGER.debugf(
+                                    "Expression part [%s] excluded from validation of [%s] against type [%s]",
+                                    info.value,
+                                    expression.toOriginalString(), match.type());
```

Comment:
I think you should condition this logger call as `toOriginalString()` is doing some allocations that we could avoid if debug is disabled.

Reply:
Well, I'm not quite sure it's necessary to save a few allocations during the build. I have a bad feeling whenever I use `Logger.isDebugEnabled()` in similar situation ;-). Also this PR did not change anything in this regard - the log message has been there a long time.

Reply:
OK I still think it's a bad practice. But I won't stop that PR for that.

Reply:
Ideally, a log method should also accept `Supplier<String>` so that we could use a method reference like `LOG.debugf("...", expression::toOriginalString)`. The `if(LOG.isDebugEnabled()) LOG.debugf(...);` construct is IMO wrong...